alternate_test.go

```diff
@@ -13,22 +12,42 @@ import (
 	"github.com/miekg/dns"
 )
 
-// NewForward Creates server and forward plugin which forwards requests to server.
-// Server provides response set by rcode. Server and forward must be disabled
-// by method Close() after use.
-func NewDummyForward(rcode int) (*dnstest.Server, *forward.Forward) {
-	s := dnstest.NewServer(func(w dns.ResponseWriter, r *dns.Msg) {
+type forwardWithServer struct {
+	s           *dnstest.Server
+	f           *forward.Forward
+	called      int
+	lastIsEdns0 bool
+}
+
+// newServer sets up http server and forward plugin which uses the test server.
+// After test finished Close() must be called to stop http server.
+func newServer(rcode int) *forwardWithServer {
```

Comment:
`newServer` should really be `newForward` because it is used as a Forward - it just happen to have a Server for testing.

Reply:
Another high level comment...

In that `rule` object we can just have an `plugin.Handler` (interface) instead of a Forward (which is a concrete object). That will make it more easily to create a mock object for testing purposes.